api/messengertypes.proto

```diff
@@ -810,3 +809,12 @@ message AudioPreview {
   uint32 bitrate = 4;
   uint32 sampling_rate = 5;
 }
+
+message Reaction {
+  string target_cid = 2 [(gogoproto.moretags) = "gorm:\"column:target_cid;primaryKey\"", (gogoproto.customname) = "TargetCID"];
+  string member_public_key = 3 [(gogoproto.moretags) = "gorm:\"primaryKey\""];
+  string emoji = 4 [(gogoproto.moretags) = "gorm:\"primaryKey\""];
+  bool is_mine = 5;
+  bool state = 6;
+  int64 state_date = 7;
+}
```

Comment:
I think you has a problem with composite primary keys in an earlier PR. Maybe you can just use a cid as the pk and wrap write on this table with transactions to remove previous data.

Can we remove the state field and delete entries if they are no longer required instead?
Can we use sent_date for consistency with other models?
Can we remove is_mine from models and handle this with a value provided globally?

Reply:
- state_date is specific to the last time the state was set so I prefer this way
- I agree but it will be a tricky query and and optimization so we can do it later